### PR TITLE
bug: fix input types for nobunaga and minipipe

### LINKED ISSUE #13: nobunaga and minipipe use incorrect input type
**Bug Description**
The `nobunaga` and `minipipe` methods use `FeatureData[Sequence]` as the input type, but expect this to be in `QIIME1DemuxDirFmt`. I think the input type should instead be `SampleData[Sequences]` (and maybe also `SampleData[SequencesWithQuality]` for convenience). 

**Steps to reproduce the behavior**
The test data contained in the q2_shogun package shows the expected format:

```
$pwd
q2-shogun/q2_shogun/tests/data
$ qiime tools export --input-path query.qza --output-path query
Exported query.qza as DNASequencesDirectoryFormat to directory query
$ head query/dna-sequences.fasta
>sample2_0 NC_013792.1_111986_1_0_0_0_0_0:0:0_0:0:0_0/1
ACTAGACTTCCGGGAGCATGACGTAGCAAGGGCTCCCCTTGTTTAGTTAT
>sample2_1 NC_013792.1_254037_1_0_0_0_0_1:0:0_0:0:0_1/1
ATAAATTCTTCAAGTTGTCTTTCTTCATGAGATTTTGCCGTGGGCTTCATT
>sample2_2 NC_013792.1_51234_1_1_1_0_0_2:0:0_0:0:0_2/1
TACATTCCTCCAGCGGTTAAAGCGGTAGAAATATCCAAAGAAAGCTGAGG
>sample2_3 NC_013792.1_223791_1_1_1_0_0_2:0:0_0:0:0_3/1
TTGTGTGTATTAGCATGGTGTGTAAACACTGTAAATAACGAAGCAACTTCG
>sample2_4 NC_013792.1_101839_1_0_0_0_0_0:0:0_0:0:0_4/1
TGTCACCGTGGTCCATTTCACTCATTTCACCCATGTCCATGTCACCATGA
```

If you then run their example command (paths modified from the description in the QIIME 2 Library, but otherwise the same command), you end up with a `FeatureTable[Frequency]` that contains multiple samples. 

```
$ qiime shogun nobunaga \
    --i-query query.qza \
    --i-reference-reads refseqs.qza \
    --i-reference-taxonomy taxonomy.qza \
    --i-database bt2-database.qza \
    --o-taxa-table taxatable.qza
$ qiime tools export --input-path taxatable.qza --output-path taxatable
$ biom summarize-table -i taxatable/feature-table.biom
Num samples: 3
Num observations: 6
...
```

**Expected behavior**
We should update this to use the correct type, or it'll be very challenging to support users as they try to prepare their data for use with these commands.

### PR DESCRIPTION
**Pull request by gregcaporaso**
The first commit in this PR fixes the input type for the two actions (fixes #13). 

The second commit adds a test for the `minipipe` action, which didn't exist before. The test fails though, and seems to be related to an issue in shogun itself (version 1.0.6, conda installed with the command `conda install -c bioconda -c conda-forge -c knights-lab shogun`). Here's a snip of the output that illustrates the issue:

```
Traceback (most recent call last):
  File "/home/jgc53/.conda/envs/qiime2-dev/bin/shogun", line 11, in <module>
    load_entry_point('shogun==1.0.6', 'console_scripts', 'shogun')()
  File "/home/jgc53/.conda/envs/qiime2-dev/lib/python3.6/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/home/jgc53/.conda/envs/qiime2-dev/lib/python3.6/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/home/jgc53/.conda/envs/qiime2-dev/lib/python3.6/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/jgc53/.conda/envs/qiime2-dev/lib/python3.6/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/jgc53/.conda/envs/qiime2-dev/lib/python3.6/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/home/jgc53/.conda/envs/qiime2-dev/lib/python3.6/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/jgc53/.conda/envs/qiime2-dev/lib/python3.6/site-packages/shogun-1.0.6-py3.6.egg/shogun/__main__.py", line 119, in pipeline
  File "/home/jgc53/.conda/envs/qiime2-dev/lib/python3.6/site-packages/shogun-1.0.6-py3.6.egg/shogun/__main__.py", line 145, in _redistribute
KeyError: 'shear'
```

Before merging, we'll either need to get this fixed upstream, or we should drop the `minipipe` action (that might be the better option as it is still insufficiently tested, even after my addition of test).

### CODE REVIEW

File: q2_shogun/plugin_setup.py [hunk 1]
```diff
@@ -39,15 +40,17 @@
 
 plugin.methods.register_function(
     function=nobunaga,
-    inputs={'query': SampleData[Sequences],
+    inputs={'query': (SampleData[Sequences] |
+                      SampleData[SequencesWithQuality] |
```
Comment (thermokarst) [inline q2_shogun/plugin_setup.py:45]: This won't work given the current view type annotation in the `nobunaga` method:

https://github.com/qiime2/q2-shogun/blob/fa1a37214f4d4b17b1394434289327f96dc0ce6b/q2_shogun/_shogun.py#L58

We don't have any transformers that would cover all of these cases (which would be a bit of a hack), but what we really need is `typing.Union` support in the framework ([see here for more discussion](https://github.com/qiime2/q2-deblur/pull/59)).

The same applies to the `minipipe` changes, below.

Comment (gregcaporaso) [reply q2_shogun/plugin_setup.py:45]: Thanks @thermokarst! I actually meant to update the view type to `QIIME1DemuxDirFmt`, (which is what shogun actually requires as input) but forgot (testing of the alternative inputs is still pending). After that change, this mirrors the input to [`dereplicate-sequences` in `q2-vsearch`](https://github.com/qiime2/q2-vsearch/blob/0f5765bc70395a961f8ed0ebb6ab3e2ba1ee7079/q2_vsearch/plugin_setup.py#L230).

Comment (thermokarst) [reply q2_shogun/plugin_setup.py:45]: Looks like I didn't see the whole changeset - the [change](https://github.com/qiime2/q2-shogun/pull/15/files#diff-0f1eefd0d2b4512bbea18fa1ac76bfaeR59) from `DNAFASTAFormat` to `QIIME1DemuxFormat` above has you set!

File: q2_shogun/_shogun.py [hunk 2]
```diff
@@ -65,7 +65,8 @@ def nobunaga(query: QIIME1DemuxFormat, reference_reads: DNAFASTAFormat,
                            database, reference_reads, reference_taxonomy)
 
         # run aligner
-        cmd = ['shogun', 'align', '-i', str(query), '-d', tmpdir,
+        query_fp = '%s/seqs.fna' % str(query)
```
Comment (thermokarst) [inline q2_shogun/_shogun.py:68]: Did you run into problems using `QIIME1DemuxFormat`? This approach is fragile (what if the dirfmt changes the file name from `seqs.fna`?), you can shore it up using:

```python
query_fp = str(query.file.view(QIIME1DemuxFormat))
```

But, I think just using the `QIIME1DemuxFormat` would be preferable (unless you ran into an issue, in which case let us know!).

Comment (gregcaporaso) [reply q2_shogun/_shogun.py:68]: Yeah, the issue I ran into was that transformers don't exist going to `QIIME1DemuxFormat`, only to `QIIME1DemuxDirFmt` ([eg](https://github.com/qiime2/q2-types/blob/261cb96f2c7af2c671e6e1a1011f0502b7604817/q2_types/per_sample_sequences/_transformer.py#L360-L387) - I had missed that before).

Comment (thermokarst) [reply q2_shogun/_shogun.py:68]: Oh yeah, that makes sense. Can you make the [change I suggested above](https://github.com/qiime2/q2-shogun/pull/15#discussion_r392316258), in that case?

Comment (gregcaporaso) [reply q2_shogun/_shogun.py:68]: Will do, thanks!

**Pull request closed without merging**